File: source/_codes/java_review/BubbleSort.java
package vell.bibi.sort_algorithms;

import vell.bibi.sort_algorithms.util.vell001;

public class BubbleSort {

	public static void sort(int[] a){
		int i, j, cup;
		boolean flag;
		for(i=a.length, flag=true; flag && i>0; i--){
			flag = false;
			for(j=0; j<i-1; j++){
				if(a[j] > a[j+1]){
					cup = a[j];
					a[j] = a[j+1];
					a[j+1] = cup;
					flag = true;
				}
			}
		}
	}
	
	public static void main(String[] args) {
		int[] a = vell001.getRandomList(10, 100);
		vell001.printList(a);
		sort(a);
		vell001.printList(a);
	}

}

File: source/_codes/c_learning/bubblesort.c
#include "vell001.h"

#define NUM 20

int* BubbleSort(int* a, int n){
	int i = 0, j = 0, flag = 1, cup = 0;
	for(i=n-1; i>0 && flag; i--){
		flag = 0; // 设置一个标记，当顺序已经排好后不再运行
		for(j=0; j<i; j++){
			if(a[j] > a[j+1]){
				cup = a[j];
				a[j] = a[j+1];
				a[j+1] = cup;
				flag = 1;
			}
		}
	}
	return a;
}

void main(){
	int* a = GetRandomNum(NUM);
	PrintList(a, NUM);
	BubbleSort(a, NUM);
	PrintList(a, NUM);
}